torchvision/prototype/datasets/_builtin/imagenet.py

```diff
@@ -201,3 +205,47 @@ def _generate_categories(self, root: pathlib.Path) -> List[Tuple[str, ...]]:
         categories_and_wnids = cast(List[Tuple[str, ...]], next(iter(meta_dp)))
         categories_and_wnids.sort(key=lambda category_and_wnid: category_and_wnid[1])
         return categories_and_wnids
+
+
+NAME = "imagenet"
+
+
+@register_info(NAME)
+def _info() -> Dict[str, Any]:
+    categories, wnids = zip(*DatasetInfo.read_categories_file(BUILTIN_DIR / f"{NAME}.categories"))
+    return dict(categories=categories, wnids=wnids)
+
+
+@register_dataset(NAME)
+class ImageNet2(Dataset2):
+    def __init__(self, root: Union[str, pathlib.Path], *, split: str = "train") -> None:
+        if split not in {"train", "val", "test"}:
+            raise ValueError
+        self._split = split
+
+        info = _info()
+        categories, wnids = info["categories"], info["wnids"]
+
+        self._old_style_dataset = ImageNet()
+        self._old_style_config = self._old_style_dataset.info.make_config(split=self._split)
+
+        self.categories = categories
+        self.info = SimpleNamespace(
+            wnid_to_category=zip(wnids, categories),
+            category_to_wnid=zip(categories, wnids),
+        )
```

Comment:
Do I understand correctly that we're only setting `self.info` for `_datapipe` to work, and that we'll remove it soon?